packages/mu-http/driver.js

```diff
@@ -14,4 +14,144 @@
 
 'use strict'
 
-// TODO
+var http = require('http')
+var stringify = require('fast-safe-stringify')
+var parse = require('fast-json-parse')
+var assert = require('assert')
+var mue = require('mu-error')()
+
+
+/**
+ * HTTP transport
+ * Figure out path mapping - perhaps this may need to be part of the config for now just /
+ */
+module.exports = function createTcpDriver (options) {
+  return function httpDriver (opts, receive) {
+    var connections = {}
+    var connectionsByIp = {}
+    var server
+
+    assert(opts, 'transport should always pass opts to httpDriver')
+    assert(receive instanceof Function, 'transport should always pass receive function to httpDriver')
+
+
+    function send (message, cb) {
+      var sendTo
+
+      if (connections[message.protocol.dst]) {
+
+        // server responding to client initaiated connection
+        if (connections[message.protocol.dst].length > 0) {
+          sendTo = connections[message.protocol.dst].pop()
+          sendTo.response.writeHead(200, { 'Content-Type': 'application/json' })
+          sendTo.response.write(stringify(message))
+          sendTo.response.end()
+        } else {
+          cb && cb(mue.transport('connection has alredy recieved response!'))
```

Comment:
Then it fails silently of course... :) actually that check was put in place whilst I was working on the last merge - will remove it and assume that the callback is always there this is a safe assumption as driver is an internal thing. Without the check it will blow up with no callback which is the correct behaviour